### PR TITLE
Remove multiple "odo app list" entries

### PR DESCRIPTION
**Pull request by cdrage**
This PR puts the app list through a "RemoveDuplicates" function to
remove any duplicate application names.

This is required since we don't have any unique identifies for
applications and thus if we have 2+ components in an application,
multiple deployment config's are returned from OpenShift with the same
application name.

See below for an example of this change:

```sh

▶ odo app list
The project 'foo' has the following applications:
NAME
foobar
app

▶ oc get dc
NAME                 REVISION   DESIRED   CURRENT   TRIGGERED BY
foobar-app           1          1         1         config,image(nodejs:latest)
java-fwwm-app        1          1         1         config,image(java:latest)
nodejs-trym-foobar   1          1         1         config,image(nodejs:latest)

▶ odo app list -o json | jq
{
  "kind": "List",
  "apiVersion": "odo.openshift.io/v1alpha1",
  "metadata": {},
  "items": [
    {
      "kind": "app",
      "apiVersion": "odo.openshift.io/v1alpha1",
      "metadata": {
        "name": "foobar",
        "namespace": "foo",
        "creationTimestamp": null
      },
      "spec": {
        "components": [
          "nodejs-trym"
        ]
      },
      "status": {
        "active": false
      }
    },
    {
      "kind": "app",
      "apiVersion": "odo.openshift.io/v1alpha1",
      "metadata": {
        "name": "app",
        "namespace": "foo",
        "creationTimestamp": null
      },
      "spec": {
        "components": [
          "foobar",
          "java-fwwm"
        ]
      },
      "status": {
        "active": false
      }
    }
  ]
}

```

Closes https://github.com/openshift/odo/issues/1713

### CODE REVIEW

**Comment by surajnarwade** [comment]
```
odoCmpSubE2e Test odo push with --source and --config flags when --context is used 
  create local nodejs component and push source and code separately
  /go/src/github.com/openshift/odo/tests/integration/component.go:203
Created dir: /tmp/123696713
Creating a new project: sfagnrghgi
Running odo with args: [project create sfagnrghgi -w] 
[odo]  •  Waiting for project to come up  ...
[odo]  ✓  Waiting for project to come up
[odo]  ✓  Project 'sfagnrghgi' is ready for use
[odo]  ✓  New project created and now using project : sfagnrghgi
Running odo with args: [component create nodejs nodejs --context /tmp/123696713 --app nodejs-push-context-test --project sfagnrghgi] 
[odo]  •  Checking component  ...
[odo]  ✓  Checking component
[odo]  •  Checking component version  ...
[odo]  ✓  Checking component version
[odo] Please use `odo push` command to create the component with source deployed
Running odo with args: [push --source --context /tmp/123696713] 
[odo]  •  Checking component  ...
[odo]  ✓  Checking component
[odo]  •  Checking component version  ...
[odo]  ✓  Checking component version
[odo]  ✗  Component nodejs does not exist and hence cannot push only source. Please use `odo push` without any flags or with both `--source` and `--config` flags
Running odo with args: [push --config --context /tmp/123696713] 
[odo]  •  Checking component  ...
[odo]  ✓  Checking component
[odo]  •  Checking component version  ...
[odo]  ✓  Checking component version
[odo]  ✓  Creating nodejs component with name nodejs
[odo]  ✓  Initializing 'nodejs' component
[odo]  •  Creating component nodejs  ...
[odo]  ✓  Creating component nodejs
[odo]  ✓  Successfully created component nodejs
[odo]  •  Applying component settings to component: nodejs  ...
[odo]  ✓  Applying component settings to component: nodejs
[odo]  ✓  Successfully updated component with name: nodejs
Running oc with args: [get dc nodejs-nodejs-push-context-test --namespace sfagnrghgi] 
[oc] NAME                              REVISION   DESIRED   CURRENT   TRIGGERED BY
[oc] nodejs-nodejs-push-context-test   1          1         0         config,image(nodejs:latest)
Running odo with args: [push --source --context /tmp/123696713] 
[odo]  ✗  Unable to connect to OpenShift cluster, is it down?
Deleting project: sfagnrghgi
```

/retest

File: pkg/util/util.go [hunk 1]
```diff
@@ -583,3 +583,24 @@ func MachineOutput(outputFlag string, resource interface{}) (string, error) {
...
 }
+
+// RemoveDuplicates goes through a string slice and removes all duplicates.
+// Reference: https://siongui.github.io/2018/04/14/go-remove-duplicates-from-slice-or-array/
+func RemoveDuplicates(s []string) []string {
+
+	// Make a map and go through each value to see if it's a duplicate or not
+	m := make(map[string]bool)
+	for _, item := range s {
+		if _, ok := m[item]; ok {
```
Comment (mik-dass) [inline pkg/util/util.go]: The if condition looks blank, why not `if _, ok :=m[item];!ok`? That way a else part won't be required too

**Comment by amitkrout** [comment]
```
[odo]  •  Waiting for project to come up  ...
Deleting project: 
```
/test integration

File: pkg/util/util.go (hunk 1 above)
Comment (cdrage) [reply pkg/util/util.go]: No idea, it's part of the code from stackoverflow :) This logic would also work too. But I think this is okay for now.

Comment (cdrage) [reply pkg/util/util.go]: Resolved!

**Comment by mik-dass** [comment]
```
[odo]  •  Waiting for project to come up  ...
Deleting project: awtupygzqx
Running odo with args: [project delete awtupygzqx -f] 
[odo]  ✗  The project awtupygzqx does not exist. Please check the list of projects using `odo project list`
```
/retest

**Pull request merged**